### ISSUE TITLE
curl certificate issues

### ISSUE THREAD

**Issue by asmeurer** [ContinuumIO/anaconda-issues#72]
_Issue by **[karissamck](https://github.com/karissamck)** from Friday Oct 11, 2013 at 16:47 GMT_
_Originally opened as https://github.com/conda/conda/issues/309_

---

I'm on Mac OSX 10.8.3

I used conda to install curl

I get an SSL error when trying to curl https websites

**Comment by chekunkov** [comment]
Very, very confusing problem... Got this in my Ubuntu 14.04. I realised it's related to anaconda distribution only after

```
$ curl-config --cflags
-I/home/vagrant/anaconda/include
```

would be nice to have this fixed. AFAIK `/etc/pki/` is used on RedHat and shouldn't be used on other platforms.

**Comment by yosuah** [comment]
I can confirm this problem can come up in plenty of different ways. I have faced it while trying to install R packages from GitHub on a Debian-based system. The package r-rcurl depends on Conda's version of curl, so it was automatically installed on the system and removing it is not an option. When trying to install any R package from GitHub (in R's shell), the following error message was presented: 

```
> library('devtools'); install_github('ropensci/plotly');
Downloading github repo ropensci/plotly@master
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Problem with the SSL CA cert (path? access rights?)
Calls: install_github ... request_fetch -> request_fetch.write_memory -> <Anonymous> -> .Call
Execution halted
```

As a temporary solution creating a symlink from Debian's default path to the one the Conda curl package expects does solve the issue:
```
mkdir -p /etc/pki/tls/certs/ \
	&& ln -s /etc/ssl/certs/ca-certificates.crt /etc/pki/tls/certs/ca-bundle.crt
```
A permanent fix for this would be much appreciated though, as hunting down this issue is non-trivial.

**Comment by dpryan79** [comment]
I'll add a quick :+1: to this. I'm trying to use pyBigWig with anaconda and am running into this issue, since `/etc/pki/tls/certs/ca-bundle.crt` doesn't exist on Ubuntu (thus, a nosetest fails and remote files can't be accessed). I would ideally like to get pyBigWig included in bioconda, but if curl is problematic then this is a non-starter.

**Comment by dpryan79** [comment]
A `build.sh` like the following appears to fix this issue:

    mkdir -p $PREFIX/etc/pki/tls/certs
    #use system curl, or whatever comes with anaconda to get the CA certificates
    curl http://curl.haxx.se/ca/cacert.pem -o $PREFIX/etc/pki/tls/certs/cacert.pem

    #Actually install curl over the broken anaconda version
    ./configure --prefix=$PREFIX --disable-ldap --with-ssl=$PREFIX --with-zlib=$PREFIX --with-ca-path=$PREFIX/etc/pki/tls/certs --with-ca-bundle=$PREFIX/etc/pki/tls/certs/cacert.pem
    make
    make install

I've made a package with that in the recipe and it seems to then work. Presumably the system curl would end up getting used to do the downloading.

**Comment by dpryan79** [comment]
At least under Linux I think this is fixed in bioconda with the "bioconda_curl" package. I've not yet looked into OSX, but since I need curl to work properly there as well I'll make that package work there if needed. The changes are quite simple and it'd be preferable if someone just confirmed that this fixes the issue and modified the curl package that comes with anaconda accordingly.

**Comment by jlmenut** [comment]
I confirm to have this problem with the latest OpenSuse Leap 42.1. A solution for me was to set   environment variable CURL_CA_BUNDLE to the good path.

`export CURL_CA_BUNDLE=/etc/ssl/ca-bundle.pem`

**Comment by goanpeca** [comment]
Closing for lack of activity.

Please reopen if this is still a problem with an update of the status.

Thanks!

**Comment by essicolo** [comment]
Pop!_OS 18.04

```
devtools::install_github("ropensci/weathercan") 
```
returns

```
Installation failed: error setting certificate verify locations:
  CAfile: microsoft-r-cacert.pem
  CApath: none
```

The [simlink solution](https://github.com/ContinuumIO/anaconda-issues/issues/72#issuecomment-141084981) did not work.

**Comment by zbjornson** [comment]
Cross-posting some notes I posted to jeroen/curl: https://github.com/jeroen/curl/issue_comments#issuecomment-411859250

It seems like Microsoft R Open or Anaconda is stripping the correct, absolute path and just using the filename (relative). I'm still digging to see where that's happening, but perhaps someone here knows where to look?

**Comment by D-K-E** [comment]
Problem still persists I am having difficulties of clonning github repos with https, because the certificate lookup location in anaconda curl is not the default certificate directory in ubuntu 16.04. 

It is extremely tiresome to work with it because most other package managers also use curl and git for managing packages and repos.

As temporary solution, anaconda installer can prompt user to import/copy certificates from their default location to anaconda specific location.

**Comment by agungor2** [comment]
my issue was: 

curl: (77) error setting certificate verify locations: CAfile: ~/anaconda3/ssl/cacert.pem CApath: none

I solved it by: 
conda install ca-certificates
